[WGSL] All passes should return after encountering the first error

tadeuzagallo opened the pull request:
#### 77b16d1e948b6779eabd8f7ef1ae984688d49a78
<pre>
[WGSL] All passes should return after encountering the first error
<a href="https://bugs.webkit.org/show_bug.cgi?id=296373">https://bugs.webkit.org/show_bug.cgi?id=296373</a>
<a href="https://rdar.apple.com/156479254">rdar://156479254</a>

Reviewed by Mike Wyrzykowski.

In 296795@main the type checker was converted to return after the first error,
this patch continues that work and changes the remaining passes (attribute
validation and visibility validation) to have the same behavior.

* Source/WebGPU/WGSL/AttributeValidator.cpp:
(WGSL::AttributeValidator::validate):
(WGSL::AttributeValidator::visit):
(WGSL::AttributeValidator::validateAlignment):
(WGSL::AttributeValidator::parseBuiltin):
(WGSL::AttributeValidator::parseInterpolate):
(WGSL::AttributeValidator::parseInvariant):
(WGSL::AttributeValidator::parseLocation):
(WGSL::AttributeValidator::validateInterpolation):
(WGSL::AttributeValidator::validateInvariant):
(WGSL::AttributeValidator::update):
(WGSL::AttributeValidator::set):
(WGSL::AttributeValidator::error):
(WGSL::AttributeValidator::validateIO):
(WGSL::AttributeValidator::validateBuiltinIO):
(WGSL::AttributeValidator::validateLocationIO):
(WGSL::AttributeValidator::validateStructIO):
* Source/WebGPU/WGSL/VisibilityValidator.cpp:
(WGSL::VisibilityValidator::validate):
(WGSL::VisibilityValidator::visit):
(WGSL::VisibilityValidator::error):
* Tools/TestWebKitAPI/Tests/WGSL/TypeCheckingTests.mm:
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Attributes)):

Canonical link: <a href="https://commits.webkit.org/297800@main">https://commits.webkit.org/297800@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7b264a6d962178f55152682548ceea811215fc9e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/112835 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/32570 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/23048 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/119038 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/63355 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/114797 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/33222 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/41133 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/85871 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/36520 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/115782 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26514 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/101504 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/66177 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/25802 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/19636 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/62797 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/95908 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/19710 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/122260 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/39913 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/29758 "Found 2 new test failures: http/tests/webgpu/webgpu/shader/validation/shader_io/locations.html http/tests/webgpu/webgpu/shader/validation/shader_io/pipeline_stage.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/94734 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/40297 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/97724 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/94473 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24134 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39599 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17413 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/36016 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/39799 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/45297 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/39438 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/42772 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/41177 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7b264a6d962178f55152682548ceea811215fc9e. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48416

tadeuzagallo committed:
[WGSL] All passes should return after encountering the first error
https://bugs.webkit.org/show_bug.cgi?id=296373
rdar://156479254

Reviewed by Mike Wyrzykowski.

In 296795@main the type checker was converted to return after the first error,
this patch continues that work and changes the remaining passes (attribute
validation and visibility validation) to have the same behavior.

* Source/WebGPU/WGSL/AttributeValidator.cpp:
(WGSL::AttributeValidator::validate):
(WGSL::AttributeValidator::visit):
(WGSL::AttributeValidator::validateAlignment):
(WGSL::AttributeValidator::parseBuiltin):
(WGSL::AttributeValidator::parseInterpolate):
(WGSL::AttributeValidator::parseInvariant):
(WGSL::AttributeValidator::parseLocation):
(WGSL::AttributeValidator::validateInterpolation):
(WGSL::AttributeValidator::validateInvariant):
(WGSL::AttributeValidator::update):
(WGSL::AttributeValidator::set):
(WGSL::AttributeValidator::error):
(WGSL::AttributeValidator::validateIO):
(WGSL::AttributeValidator::validateBuiltinIO):
(WGSL::AttributeValidator::validateLocationIO):
(WGSL::AttributeValidator::validateStructIO):
* Source/WebGPU/WGSL/VisibilityValidator.cpp:
(WGSL::VisibilityValidator::validate):
(WGSL::VisibilityValidator::visit):
(WGSL::VisibilityValidator::error):
* Tools/TestWebKitAPI/Tests/WGSL/TypeCheckingTests.mm:
(TestWGSLAPI::TEST(WGSLTypeCheckingTests, Attributes)):

Canonical link: https://commits.webkit.org/297800@main
